erasure-readfile.go

```diff
@@ -25,6 +25,60 @@ import (
 	"github.com/klauspost/reedsolomon"
 )
 
+// isSuccessDecodeBlocks - do we have all the blocks to be successfully decoded?.
+func isSuccessDecodeBlocks(disks []StorageAPI, dataBlocks int) bool {
+	// Count number of data and parity blocks that were read.
+	var successDataBlocksCount = 0
+	var successParityBlocksCount = 0
+	for index, disk := range disks {
+		if disk == nil {
+			continue
+		}
+		if index < dataBlocks {
+			successDataBlocksCount++
+			continue
+		}
+		successParityBlocksCount++
+	}
+	return successDataBlocksCount+successParityBlocksCount >= dataBlocks+1
+}
+
+// isSuccessDataBlocks - do we have all the data blocks?
+func isSuccessDataBlocks(disks []StorageAPI, dataBlocks int) bool {
+	// Count number of data and parity blocks that were read.
```

Comment:
This comment doesn't add up. We are counting available data blocks in list of disks. We are not counting parity blocks here.